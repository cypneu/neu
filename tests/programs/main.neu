// Arithmetic
x = 5 * (3 + 4) % 3;
y = 1234.45;
z = x + y;

// String ops
s1 = "asdf";
s2 = "fdas";
s3 = s1 + s2;

// Boolean
a = true;
b = false;

// Scope
xouter = 5;
{
  xinner1 = 4;
  {
    xinner2 = xouter + 1;
    xinner1 = 5;
  }
  x3 = xinner1 + xouter;
}

// Branching control flow
if true {
  branchTrue = "ran-true-branch";
}

if false {
  branchSkipped = "you shouldn't see this";
}

if x > 0 {
  positiveBranch = true;
} else {
  positiveBranch = false;
}

if x == 2 {
  nestedA = "x-is-two";
  if y < 1000 {
    nestedB = "y-small";
  } else if y < 1300 {
    nestedB = "y-medium";
  } else {
    nestedB = "y-large";
  }
} else {
  nestedA = "x-not-two";
  nestedB = "no-inner";
}
