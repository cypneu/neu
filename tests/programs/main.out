2 1234.45 1236.45
asdf fdas asdffdas
5
Hello
Positive
x-is-two
y-medium
logical and comparison true
3 6
10
1
2
